### PR TITLE
Adding a Jenkinsfile

### PR DESCRIPTION
**Pull request by rsandell**

### CODE REVIEW

**Comment by rsandell** [comment]
@reviewbybees

**Comment by rtyler** [comment]
declarative Pipeline even, you're so clever :smiley_cat: 


Does this require any special version of the model definition plugin, or will a recent one work fine?

**Comment by rsandell** [comment]
@rtyler This should work with whatever model definition we have installed since I based it on the Jenkinsfile in the PMD plugin and that seems to be running fine  ;)

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+pipeline {
+    agent label:"docker", docker: "maven:3.3.9-jdk-7"
```
Comment (jglick) [inline Jenkinsfile]: Prefer to build everything on 8 unless there is a specific reason to use 7.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    environment {
+      MAVEN_OPTS = "-XX:MaxPermSize=256m -Xmx1024m"
```
Comment (jglick) [inline Jenkinsfile]: Of course the `MaxPermSize` would be unnecessary on 8. Whether `Mx` is really needed here is TBD. Offhand I would guess that the JRE’s ergonomics are blind to the fact that you are running in a container, and thus the default max heap is going to vary according to host environment.

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    }
+
+    stages {
+        stage("Build") {
+            sh 'mvn clean install -Dmaven.test.failure.ignore=true'
```
Comment (jglick) [inline Jenkinsfile]: `=true` can be omitted.

Comment (jglick) [inline Jenkinsfile]: And always pass `-B`.

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        }
+    }
+
+    postBuild {
+        always {
```
Comment (jglick) [inline Jenkinsfile]: Even if the build fails? Will this not pick up artifacts from an older build using the same workspace?

Comment (jglick) [inline Jenkinsfile]: Or cause a spurious and confusing failure in the post-build section after a compilation failure? I would think you would want to do this only if the main build was a success.

File: Jenkinsfile (hunk 1 above)
Comment (rsandell) [inline Jenkinsfile]: Done

File: Jenkinsfile (hunk 2 above)
Comment (rsandell) [inline Jenkinsfile]: Done

File: Jenkinsfile (hunk 3 above)
Comment (rsandell) [inline Jenkinsfile]: Yes, `=true` can be omited but, it [disturbs my circles](https://en.wikipedia.org/wiki/Noli_turbare_circulos_meos!) ;)

Added `-B`

File: Jenkinsfile (hunk 4 above)
Comment (rsandell) [inline Jenkinsfile]: Done

**Comment by stephenc** [comment]
:bee:

File: Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,21 @@
+pipeline {
+    agent label:"docker", docker: "maven:3.3.9-jdk-8"
+
+    environment {
+      MAVEN_OPTS = "-Xmx1024m"
+    }
+
+    stages {
+        stage("Build") {
+            sh 'mvn clean install -B -Dmaven.test.failure.ignore=true'
```
Comment (rsandell) [inline Jenkinsfile]: Just pure habit

**Pull request merged**